src/Rings/orderings.jl

```diff
@@ -174,6 +174,7 @@ mutable struct MonomialOrdering{S}
   o::AbsGenOrdering
   is_total::Bool
   is_total_is_known::Bool
+  canonical_matrix::ZZMatrix
 end
 
 function MonomialOrdering(R::S, o::AbsGenOrdering) where S
```

Comment:
I think you need to adjust this constructor to fix the errors: either make it an non-parametric inner constructor (so use `new{S}` in the next line); or else add a new inner parametric constructor that allows omitting the final field.